src/Server.php

```diff
@@ -226,7 +238,27 @@ public function handleRequest(ConnectionInterface $conn, RequestInterface $reque
         );
 
         $callback = $this->callback;
-        $callback($request, $response);
+        $promise = new Promise(function ($resolve, $reject) use ($callback, $request) {
+            $resolve($callback($request));
+        });
+
+        $that = $this;
+        $promise->then(
+            function ($response) use ($that, $conn, $request) {
+                if (!$response instanceof ResponseInterface) {
+                    $that->emit('error', array(new \InvalidArgumentException('Invalid response type')));
+                    return $that->writeError($conn, 500);
+                }
+                $that->handleResponse($conn, $response, $request->getProtocolVersion());
+            },
+            function ($ex) use ($that, $conn) {
+                if (!$ex instanceof \Exception) {
```

Comment:
Thanks for the review.I reconsidered this and changed the code. The `error`-event on the Server should always be an Exception.
In the newest changes a `RuntimeExpception` will always be created, and a thrown exception will be added as previous to this `RuntimeException`.
Checkout the documentation and the code.